### PR TITLE
System smoke test (includes PR #33)

### PR DESCRIPTION
**Pull request by felix-b**
PR #33 was destroyed by an accident. This is the new PR that tracks completion of PR #33.

### CODE REVIEW

**Comment by felix-b** [comment]
## Factual summary

### How to start and stop microservice in a system test

This is how starting a microservice in a system test looks:

```c#
var hostController = new MicroserviceHostBuilder(microserviceName: "SmokeTest")
	.UseCliDirectoryFromSource(relativeProjectDirectoryPath: "..", allowOverrideByEnvironmentVar: true)
	.UseMicroserviceFromSource(relativeProjectDirectoryPath: "..")
	.UseAutofacInjectionAdapter()
	.UseApplicationFeature<SmokeTestFeatureLoader>()
	.Build();

hostController.Start();
```

See https://github.com/felix-b/NWheels/blob/pr-smoke-test/Source/NWheels.SystemTests/Microservices/MicroserviceHostSmokeTest.cs for full example.

### Hot to run unit tests separately from system tests

In the code of the system test:
```c#
[Trait("Purpose", "SystemTest")]
public class MicroserviceHostSmokeTest
{
    // .........
}
```

In Visual Studio:
system-test-vs

Note that system tests have trait Purpose=SystemTest, while unit tests have no traits.

### Here is how it looks in the build log

system-test-1

Suggested future enhancements: #41, #40.

File: Source/NWheels.SystemTests/Microservices/MicroserviceHostSmokeTest.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        {
+            var hostController = new MicroserviceHostBuilder(microserviceName: "SmokeTest")
+                .UseCliDirectoryFromSource(relativeProjectDirectoryPath: "..", allowOverrideByEnvironmentVar: true)
+                .UseMicroserviceFromSource(relativeProjectDirectoryPath: "..")
+                .UseAutofacInjectionAdapter()
+                .UseApplicationFeature<SmokeTestFeatureLoader>()
+                .Build();
+
+            hostController.Start();
+
```
Comment (ILyaIL83) [inline Source/NWheels.SystemTests/Microservices/MicroserviceHostSmokeTest.cs]: On my slow machine(Core i3-2310M CPU @ 2.10GHz(Cores: 2, Logical processors: 4) with 4GB memory) Kestrel doesn't have enough time to start until I add "System.Threading.Thread.Sleep(1000);"
[SystemTestError.txt](https://github.com/felix-b/NWheels/files/910491/SystemTestError.txt)

Comment (felix-b) [inline Source/NWheels.SystemTests/Microservices/MicroserviceHostSmokeTest.cs]: Oops. Thread.Sleep isn't good, either. We should wait until it prints "Microservice is up". Or signals a global mutex. I'll open a bug on this.

Comment (felix-b) [inline Source/NWheels.SystemTests/Microservices/MicroserviceHostSmokeTest.cs]: Ha ha, it has just failed the build :D opened bug #47. Will fix it shortly.

**Pull request merged**